libdevice/include/asan_libdevice.hpp

```diff
@@ -0,0 +1,164 @@
+//==-- device-sanitizer-report.hpp - Structure and declaration for assert
+// support --==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+#pragma once
+
+#include <cinttypes>
+
+#if OMP_LIBDEVICE && INTEL_COLLAB
+#pragma omp declare target
+#endif
+
+// NOTE This file should be sync with
+// unified-runtime/source/loader/layers/sanitizer/device_sanitizer_report.hpp
```

Comment:
Why can't we make it a public header in UR and just include it here?

Reply:
Hi, @aelovikov-intel 
When building the whole project, clang is firstly built and libdevice is built using the fresh built clang, at that time, sycl/ur component hasn't been built and deployed, so we can only include standard c/c++ headers in libdevice.
Thanks very much.

Reply:
Why can't we change that?

Reply:
Hi, @aelovikov-intel 
Since libdevice component is shared by sycl and omp and libdeivce is a parallel component with sycl and omp , so we intend to minimize its dependency on any sycl/omp specific header files.

Reply:
> its dependency on any sycl/omp specific header files

How does it apply to UR?